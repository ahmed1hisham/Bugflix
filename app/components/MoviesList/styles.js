import {StyleSheet} from 'react-native';
import {red, veryDarkGrey, white} from '../../theme/colors';

export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: veryDarkGrey,
    alignItems: 'center',
  },
  activityIndicatorContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  moreActivityIndicator: {
    marginTop: 20,
  },
  emptyMoviesView: {
    flex: 1,
    height: '100%',
    width: '100%',
    paddingTop: 50,
    paddingHorizontal: 15,
  },
  emptyMoviesText: {
    width: '80%',
    fontSize: 20,
    color: white,
    fontWeight: 'bold',
    textAlign: 'center',
  },
  refreshContainer: {
    marginTop: 10,
    padding: 15,
    backgroundColor: red,
    borderRadius: 10,
  },
});
