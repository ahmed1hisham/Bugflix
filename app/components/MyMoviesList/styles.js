import {StyleSheet} from 'react-native';
import {veryDarkGrey, white} from '../../theme/colors';
export const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: veryDarkGrey,
    alignItems: 'center',
  },
  activityIndicatorContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  moreActivityIndicator: {
    marginTop: 20,
  },
  emptyMoviesView: {
    flex: 1,
    height: '100%',
    width: '100%',
    paddingTop: 50,
    paddingHorizontal: 15,
  },
  emptyMoviesText: {
    width: '80%',
    fontSize: 20,
    color: white,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});
