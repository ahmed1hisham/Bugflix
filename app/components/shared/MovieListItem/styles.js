import {StyleSheet, Dimensions} from 'react-native';
import {veryDarkGrey, white, lightGrey, grey} from '../../../theme/colors';

const {width, height} = Dimensions.get('window');

export const styles = StyleSheet.create({
  movieCardContainer: {
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    width: width - 30,
    marginVertical: 7.5,
    backgroundColor: veryDarkGrey,
    borderRadius: 10,
    borderWidth: 1,
    borderColor: grey,
    shadowColor: '#000000',
    shadowOffset: {width: 0, height: 0},
    shadowOpacity: 0.16,
    shadowRadius: 6,
    elevation: 6,
    overflow: 'hidden',
    height: 200,
  },
  titleStyle: {
    fontSize: 24,
    color: white,
    fontWeight: 'bold',
    marginBottom: 5,
  },
  dateStyle: {
    fontSize: 16,
    color: lightGrey,
    marginBottom: 15,
  },
  overviewStyle: {
    width: '100%',
    flexWrap: 'wrap',
    fontSize: 16,
    color: lightGrey,
  },
  imageContainerStyle: {
    flex: 2,
    width: 135,
    height: '100%',
  },
  imageStyle: {width: 135, height: '100%', resizeMode: 'contain'},
  infoContainer: {padding: 10, flex: 3},
});
