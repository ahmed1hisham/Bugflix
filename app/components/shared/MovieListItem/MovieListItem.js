import React from 'react';
import {View, Text, Dimensions, Image, ScrollView} from 'react-native';
import {styles} from './styles';

const {width, height} = Dimensions.get('window');
const imageBaseUrl = 'https://image.tmdb.org/t/p/';
const imageFetchSize = 'w342';

const MovieListItem = (props) => {
  const {movie} = props;
  return (
    <View style={styles.movieCardContainer}>
      <View style={styles.imageContainerStyle}>
        <Image
          style={styles.imageStyle}
          source={
            movie.poster_path === null
              ? require('../../../assets/images/default_poster.jpg')
              : movie.userCreated
              ? {uri: movie.poster_path}
              : {uri: imageBaseUrl + imageFetchSize + movie.poster_path}
          }
        />
      </View>
      <View style={styles.infoContainer}>
        <Text style={styles.titleStyle}>{movie.title}</Text>
        <Text style={styles.dateStyle}>Release Date: {movie.release_date}</Text>
        <ScrollView>
          <Text style={styles.overviewStyle}>{movie.overview}</Text>
        </ScrollView>
      </View>
    </View>
  );
};

export default React.memo(MovieListItem, (prevProps, nextProps) => {
  return (
    prevProps.movie.title === nextProps.movie.title &&
    prevProps.movie.poster_path === nextProps.movie.poster_path
  );
});
