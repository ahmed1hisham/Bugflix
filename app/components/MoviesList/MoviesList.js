import React, {useState, useEffect} from 'react';
import {
  View,
  FlatList,
  Alert,
  ActivityIndicator,
  Text,
  ScrollView,
  TouchableOpacity,
  RefreshControl,
} from 'react-native';
import {white} from '../../theme/colors';
import MovieListItem from '../shared/MovieListItem/MovieListItem';
import axios from 'axios';
import {styles} from './styles';

const apiKey = 'acea91d2bff1c53e6604e4985b6989e2';
const movieBaseUrl = `http://api.themoviedb.org/3/discover/movie?api_key=${apiKey}&page=`;
const initialUrl = `${movieBaseUrl}1`;

const MoviesList = (props) => {
  const [movies, setMovies] = useState([]);
  const [refreshing, setRefreshing] = useState(false);
  const [loading, setLoading] = useState(false);
  const [fetchingMore, setFetchingMore] = useState(false);
  const [pageToFetch, setPageToFetch] = useState(1);

  useEffect(() => {
    fetchMovies();
  }, []);

  const fetchMovies = async () => {
    setLoading(true);
    await moviesApiCall();
    setLoading(false);
  };

  const onRefresh = async () => {
    setRefreshing(true);
    await moviesApiCall();
    setRefreshing(false);
  };

  const moviesApiCall = async () => {
    if (props.isConnected === true) {
      await axios
        .get(initialUrl)
        .then((response) => {
          setMovies(response.data.results);
        })
        .catch((error) => {
          Alert.alert(error);
        });
    } else {
      Alert.alert('No Internet Connection');
    }
  };

  const fetchMoreMovies = async () => {
    if (fetchingMore === false) {
      setFetchingMore(true);
      let newPage = pageToFetch + 1;
      try {
        if (props.isConnected === true) {
          await axios
            .get(movieBaseUrl + newPage)
            .then((response) => {
              let newMovies = movies.concat(response.data.results);
              setMovies(newMovies);
            })
            .catch((error) => {
              Alert.alert(error);
            });
        } else {
          Alert.alert('No Internet Connection');
        }
      } catch (error) {
        Alert.alert(error);
      }
      setFetchingMore(false);
      setPageToFetch(newPage);
    }
  };

  const renderFooter = () => {
    if (fetchingMore === true) {
      return (
        <View testID="fetchingMoreIndicator">
          <ActivityIndicator
            testID="fetchingMoreIndicator"
            style={styles.moreActivityIndicator}
            color={white}
            size="large"
          />
        </View>
      );
    } else {
      return null;
    }
  };

  const renderItem = ({item}) => (
    <MovieListItem testID="movieListItem" movie={item} />
  );

  return (
    <View testID="allMoviesListContainer" style={styles.container}>
      {props.isConnected !== true ? (
        <ScrollView
          style={styles.emptyMoviesView}
          contentContainerStyle={{alignItems: 'center'}}>
          <Text style={styles.emptyMoviesText}>No internet connection</Text>
          <TouchableOpacity
            onPress={fetchMovies}
            style={styles.refreshContainer}>
            <Text style={styles.emptyMoviesText}>Refresh</Text>
          </TouchableOpacity>
        </ScrollView>
      ) : loading === true ? (
        <View style={styles.activityIndicatorContainer}>
          <ActivityIndicator color={white} size="large" />
        </View>
      ) : movies.length > 0 ? (
        <FlatList
          testID="listOfMovies"
          style={{flex: 1}}
          style={{paddingTop: 7.5}}
          contentContainerStyle={{paddingBottom: 15}}
          data={movies}
          scrollEnabled={true}
          showsVerticalScrollIndicator={false}
          refreshControl={
            <RefreshControl
              refreshing={refreshing}
              onRefresh={onRefresh}
              tintColor="#fff"
            />
          }
          renderItem={renderItem}
          keyExtractor={(item) => {
            return item.id.toString() + pageToFetch.toString();
          }}
          onEndReached={fetchMoreMovies}
          onEndReachedThreshold={0.001}
          ListFooterComponent={renderFooter}
          nestedScrollEnabled={true}
        />
      ) : (
        <ScrollView
          style={styles.emptyMoviesView}
          contentContainerStyle={{alignItems: 'center'}}>
          <Text style={styles.emptyMoviesText}>
            No movies to show right now, come back later ;)
          </Text>
          <TouchableOpacity
            onPress={fetchMovies}
            style={styles.refreshContainer}>
            <Text style={styles.emptyMoviesText}>Refresh</Text>
          </TouchableOpacity>
        </ScrollView>
      )}
    </View>
  );
};

export default React.memo(MoviesList);
