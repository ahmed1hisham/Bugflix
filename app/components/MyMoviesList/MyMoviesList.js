import React from 'react';
import {FlatList, View, Text, ScrollView} from 'react-native';
import MovieListItem from '../shared/MovieListItem/MovieListItem';
import {styles} from './styles';

export default MyMoviesList = (props) => {
  return (
    <View testID="myMoviesListContainer" style={styles.container}>
      {props.movies.length > 0 ? (
        <FlatList
          testID="myMoviesFlatList"
          style={{flex: 1}}
          style={{paddingTop: 7.5}}
          contentContainerStyle={{paddingBottom: 15}}
          data={props.movies}
          scrollEnabled={true}
          showsVerticalScrollIndicator={false}
          renderItem={({item}) => (
            <MovieListItem testID="movieListItem" movie={item} />
          )}
          keyExtractor={(item) => {
            return item.id.toString();
          }}
        />
      ) : (
        <ScrollView
          testID="noMoviesYetContainer"
          style={styles.emptyMoviesView}
          contentContainerStyle={{alignItems: 'center'}}>
          <Text style={styles.emptyMoviesText}>
            You haven't added any movies yet, go ahead add one now!
          </Text>
        </ScrollView>
      )}
    </View>
  );
};
