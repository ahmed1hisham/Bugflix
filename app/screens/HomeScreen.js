import React, {Component, useState, useEffect} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  View,
  Dimensions,
  Text,
  Alert,
} from 'react-native';
import {veryDarkGrey, red, white} from '../theme/colors';
import BugflixTitle from '../components/shared/BugflixTitle/BugflixTitle';
import {TabView, TabBar} from 'react-native-tab-view';
import MoviesList from '../components/MoviesList/MoviesList';
import MyMoviesList from '../components/MyMoviesList/MyMoviesList';
import FAB from 'react-native-fab';
import AddMovieModal from '../components/AddMovieModal/AddMovieModal';
import OfflineMessage from '../components/shared/OfflineMessage/OfflineMessage';
import NetInfo from '@react-native-community/netinfo';

const {width, height} = Dimensions.get('window');

const HomeScreen = (props) => {
  const [index, setIndex] = useState(0);
  const [idCounter, setIdCounter] = useState(1);
  const [addMovieVisible, setAddMoviesVisible] = useState(false);
  const [myMovies, setMyMovies] = useState([]);
  const [isConnected, setIsConnected] = useState(true);
  const [navState, setNavState] = useState({
    index: 0,
    routes: [
      {key: 'movies', title: 'All Movies'},
      {key: 'myMovies', title: 'My Movies'},
    ],
  });

  useEffect(() => {
    const unsubscribe = NetInfo.addEventListener((state) => {
      setIsConnected(state.isConnected);
    });

    return () => {
      unsubscribe();
    };
  }, []);

  const checkConnection = () => {
    NetInfo.fetch().then((state) => {
      setIsConnected(state.isConnected);
      if (state.isConnected === false) {
        Alert.alert('No Internet Connection');
      }
    });
  };

  const addNewMovie = (movie) => {
    movie.id = idCounter;
    let movies = myMovies;
    movies.push(movie);
    setMyMovies(movies);
    setIdCounter(movie.id + 1);
    setNavState({...navState, index: 1});
  };

  const renderScene = ({route}) => {
    switch (route.key) {
      case 'movies':
        return <MoviesList />;
      case 'myMovies':
        return <MyMoviesList movies={myMovies} />;
      default:
        return null;
    }
  };

  return (
    <SafeAreaView style={styles.container}>
      <BugflixTitle />
      <TabView
        navigationState={navState}
        renderScene={renderScene}
        onIndexChange={(index) => {
          setNavState({...navState, index});
        }}
        initialLayout={width}
        renderTabBar={(props) => (
          <View style={{alignItems: 'center'}}>
            <TabBar
              {...props}
              style={{
                backgroundColor: veryDarkGrey,
                elevation: 0,
                width: '100%',
              }}
              indicatorStyle={{
                backgroundColor: red,
                height: 4,
              }}
              labelStyle={{
                color: white,
                fontSize: 14,
                fontWeight: '500',
                lineHeight: 20,
              }}
            />
          </View>
        )}
      />
      <FAB
        buttonColor={red}
        iconTextColor="#FFFFFF"
        onClickAction={() => {
          setAddMoviesVisible(true);
        }}
        visible={true}
        iconTextComponent={<Text style={styles.FABIconStyle}>+</Text>}
      />
      <AddMovieModal
        isVisible={addMovieVisible}
        onClose={() => {
          setAddMoviesVisible(false);
        }}
        addNewMovie={addNewMovie}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: veryDarkGrey,
  },
  FABIconStyle: {
    fontSize: 22,
    fontWeight: 'bold',
  },
});
export default HomeScreen;
