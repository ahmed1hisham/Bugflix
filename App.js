import 'react-native-gesture-handler';
import React, {Component} from 'react';
import {AppRegistry, StatusBar} from 'react-native';
import {veryDarkGrey} from './app/theme/colors';
import HomeScreen from './app/screens/HomeScreen/HomeScreen';
import Icon from 'react-native-vector-icons/MaterialIcons';
import SplashScreen from 'react-native-splash-screen';
import {LogBox} from 'react-native';

LogBox.ignoreAllLogs();

export default class App extends Component {
  componentDidMount() {
    Icon.loadFont();
    SplashScreen.hide();
  }
  render() {
    return (
      <>
        <StatusBar backgroundColor={veryDarkGrey} barStyle="light-content" />
        <HomeScreen />
      </>
    );
  }
}

AppRegistry.registerComponent('App', () => App);
