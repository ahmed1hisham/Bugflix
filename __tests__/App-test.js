import 'react-native';
import React, {useState as useStateMock} from 'react';
import App from '../App';
import renderer from 'react-test-renderer';
import HomeScreen from '../app/screens/HomeScreen/HomeScreen';
import {shallow} from 'enzyme';
import MoviesList from '../app/components/MoviesList/MoviesList';
import MyMoviesList from '../app/components/MyMoviesList/MyMoviesList';
import MovieListItem from '../app/components/shared/MovieListItem/MovieListItem';

jest.useFakeTimers();

it('App renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Home renders correctly', () => {
  const component = shallow(<HomeScreen />);
  expect(component).toMatchSnapshot();
});

it('Movies List Component renders correctly', () => {
  const component = shallow(<MoviesList isConnected={true} />);
  expect(component).toMatchSnapshot();
});

it('My Movie List Component renders correctly', () => {
  const movies = [
    {
      poster_path: null,
      userCreated: true,
      title: 'TestMovie',
      overview: 'Test overview',
      release_date: '2020',
      id: 1,
    },
  ];
  const component = shallow(<MyMoviesList movies={movies} />);
  expect(component).toMatchSnapshot();
});

it('Add movie modal renders correctly', () => {
  const component = shallow(
    <MoviesList
      isVisible={true}
      onClose={() => {
        jest.fn();
      }}
      testID="fakeTestID"
    />,
  );
  expect(component).toMatchSnapshot();
});
