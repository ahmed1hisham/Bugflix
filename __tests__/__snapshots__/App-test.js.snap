// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add movie modal renders correctly 1`] = `ShallowWrapper {}`;

exports[`App renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#131313",
      "flex": 1,
    }
  }
  testID="homeScreenContainer"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginBottom": 15,
        "marginTop": 15,
        "paddingHorizontal": 15,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../app/assets/images/Header.png",
        }
      }
      style={
        Object {
          "height": 42,
          "resizeMode": "contain",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 50,
        "bottom": 20,
        "height": 62,
        "justifyContent": "center",
        "position": "absolute",
        "right": 17,
        "width": 62,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "stretch",
          "borderRadius": 50,
          "elevation": 2,
          "height": 56,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.8,
          "shadowRadius": 2,
          "width": 56,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#E50914",
            "borderRadius": 50,
            "flex": 1,
            "justifyContent": "center",
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Object {
              "transform": Array [
                Object {
                  "scaleX": 1,
                },
                Object {
                  "rotate": "0deg",
                },
              ],
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 24,
              }
            }
            testID="actionButton"
          >
            +
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    testID="addMovieModal"
  >
    <Modal
      animationType="none"
      deviceHeight={1334}
      deviceWidth={750}
      hardwareAccelerated={false}
      hideModalContentWhileAnimating={false}
      onBackdropPress={[Function]}
      onModalHide={[Function]}
      onModalWillHide={[Function]}
      onModalWillShow={[Function]}
      onRequestClose={[Function]}
      scrollHorizontal={false}
      scrollOffset={0}
      scrollOffsetMax={0}
      scrollTo={null}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      swipeThreshold={100}
      testID="addMovieModal"
      transparent={true}
      visible={false}
    >
      <View
        accessible={true}
        focusable={true}
        forwardedRef={[Function]}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
      />
      <View
        deviceHeight={1334}
        deviceWidth={750}
        forwardedRef={[Function]}
        hideModalContentWhileAnimating={false}
        onBackdropPress={[Function]}
        onModalHide={[Function]}
        onModalWillHide={[Function]}
        onModalWillShow={[Function]}
        pointerEvents="box-none"
        scrollHorizontal={false}
        scrollOffset={0}
        scrollOffsetMax={0}
        scrollTo={null}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
        supportedOrientations={
          Array [
            "portrait",
            "landscape",
          ]
        }
        swipeThreshold={100}
      >
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "bottom": 0,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#131313",
                  "borderRadius": 10,
                  "height": 933.8,
                  "padding": 15,
                  "width": 675,
                }
              }
              testID="addMovieModal"
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                    "marginBottom": 5,
                    "width": "100%",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                />
              </View>
              <Text
                style={
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 24,
                    "fontWeight": "bold",
                    "marginBottom": 15,
                  }
                }
              >
                Enter Movie Details
              </Text>
              <RCTScrollView
                contentContainerStyle={
                  Object {
                    "alignItems": "center",
                    "paddingTop": 10,
                  }
                }
                showsVerticalScrollIndicator={false}
                style={
                  Object {
                    "height": "100%",
                    "width": "100%",
                  }
                }
              >
                <View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "justifyContent": "center",
                        "opacity": 1,
                        "width": "100%",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "height": 150,
                          "marginBottom": 15,
                          "width": 135,
                        }
                      }
                    >
                      <Image
                        source={
                          Object {
                            "testUri": "../../../app/assets/images/default_poster.jpg",
                          }
                        }
                        style={
                          Object {
                            "height": "100%",
                            "resizeMode": "contain",
                            "width": 135,
                          }
                        }
                      />
                    </View>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCompleteType="off"
                    autoCorrect={false}
                    onChangeText={[Function]}
                    placeholder="Movie Title"
                    placeholderTextColor="#BABABA"
                    rejectResponderTermination={true}
                    selectionColor="#E50914"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#333333",
                          "borderRadius": 6,
                          "color": "#FFFFFF",
                          "fontSize": 18,
                          "fontWeight": "500",
                          "height": 50,
                          "paddingHorizontal": 15,
                          "width": "100%",
                        },
                        Object {
                          "marginBottom": 10,
                        },
                        Object {},
                      ]
                    }
                    testID="movieTitle"
                    underlineColorAndroid="transparent"
                    value={null}
                  />
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "backgroundColor": "#333333",
                        "borderRadius": 6,
                        "height": 50,
                        "justifyContent": "center",
                        "marginBottom": 10,
                        "opacity": 1,
                        "paddingHorizontal": 15,
                        "width": "100%",
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "fontSize": 18,
                            "fontWeight": "500",
                          },
                          Object {
                            "color": "#BABABA",
                          },
                        ]
                      }
                      testID="datePicker"
                    >
                      Release Date
                    </Text>
                  </View>
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="none"
                    autoCompleteType="off"
                    autoCorrect={false}
                    description={true}
                    multiline={true}
                    onChangeText={[Function]}
                    placeholder="Description"
                    placeholderTextColor="#BABABA"
                    rejectResponderTermination={true}
                    selectionColor="#E50914"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "#333333",
                          "borderRadius": 6,
                          "color": "#FFFFFF",
                          "fontSize": 18,
                          "fontWeight": "500",
                          "height": 50,
                          "paddingHorizontal": 15,
                          "width": "100%",
                        },
                        Object {},
                        Object {
                          "height": 100,
                        },
                      ]
                    }
                    testID="overview"
                    underlineColorAndroid="transparent"
                    value={null}
                  />
                </View>
              </RCTScrollView>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                    "marginTop": 15,
                    "width": "100%",
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#BABABA",
                      "borderRadius": 6,
                      "justifyContent": "center",
                      "opacity": 1,
                      "padding": 15,
                      "width": 75,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#FFFFFF",
                        "fontSize": 16,
                        "fontWeight": "500",
                      }
                    }
                    testID="submitMovie"
                  >
                    Save
                  </Text>
                </View>
              </View>
            </View>
            <View
              testID="datePickerModal"
            >
              <Modal
                animationType="none"
                deviceHeight={1334}
                deviceWidth={750}
                hardwareAccelerated={false}
                hideModalContentWhileAnimating={false}
                onBackdropPress={[Function]}
                onModalHide={[Function]}
                onModalWillHide={[Function]}
                onModalWillShow={[Function]}
                onRequestClose={[Function]}
                scrollHorizontal={false}
                scrollOffset={0}
                scrollOffsetMax={0}
                scrollTo={null}
                supportedOrientations={
                  Array [
                    "portrait",
                    "landscape",
                  ]
                }
                swipeThreshold={100}
                transparent={true}
                visible={false}
              >
                <View
                  accessible={true}
                  focusable={true}
                  forwardedRef={[Function]}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "backgroundColor": "black",
                      "bottom": 0,
                      "height": 1334,
                      "left": 0,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": 750,
                    }
                  }
                />
                <View
                  deviceHeight={1334}
                  deviceWidth={750}
                  forwardedRef={[Function]}
                  hideModalContentWhileAnimating={false}
                  onBackdropPress={[Function]}
                  onModalHide={[Function]}
                  onModalWillHide={[Function]}
                  onModalWillShow={[Function]}
                  pointerEvents="box-none"
                  scrollHorizontal={false}
                  scrollOffset={0}
                  scrollOffsetMax={0}
                  scrollTo={null}
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "center",
                      "flex": 1,
                      "justifyContent": "center",
                      "margin": 37.5,
                      "transform": Array [
                        Object {
                          "translateY": 0,
                        },
                      ],
                      "zIndex": 10,
                    }
                  }
                  supportedOrientations={
                    Array [
                      "portrait",
                      "landscape",
                    ]
                  }
                  swipeThreshold={100}
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#131313",
                        "borderRadius": 10,
                        "justifyContent": "center",
                        "paddingVertical": 15,
                        "width": 630,
                      }
                    }
                  >
                    <RNDatePicker
                      date={1610555513833}
                      minuteInterval={1}
                      mode="date"
                      onChange={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "height": 216,
                            "width": 310,
                          },
                          undefined,
                        ]
                      }
                      textColor="#FFFFFF"
                    />
                    <View
                      style={
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "justifyContent": "flex-end",
                          "paddingHorizontal": 15,
                          "width": "100%",
                        }
                      }
                    >
                      <Text
                        onPress={[Function]}
                        style={
                          Object {
                            "color": "#E50914",
                            "fontSize": 18,
                            "fontWeight": "bold",
                          }
                        }
                        testID="pickingDateDone"
                      >
                        Done
                      </Text>
                    </View>
                  </View>
                </View>
              </Modal>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</RCTSafeAreaView>
`;

exports[`Home renders correctly 1`] = `ShallowWrapper {}`;

exports[`Movies List Component renders correctly 1`] = `ShallowWrapper {}`;

exports[`My Movie List Component renders correctly 1`] = `ShallowWrapper {}`;
